vertexbufferlayout.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <vector>
+
+#include "renderutils.h"
+
+struct VertexBufferElement
+{
+    unsigned int mType;
+    unsigned int mCount;
+    bool         mNormalized;
+
+    static unsigned int GetSizeOfType(unsigned int type)
+    {
+        switch (type)
+        {
+            case GL_FLOAT:          return 4;
+            case GL_UNSIGNED_INT:   return 4;
+            case GL_UNSIGNED_BYTE:  return 1;
+            default:                return 0;
+        }
+    }
+};
+
+class VertexBufferLayout
+{
+private:
+    std::vector<VertexBufferElement> mElements;
+    unsigned int mStride;
+public:
+    VertexBufferLayout();
+    ~VertexBufferLayout();
+
+    template<typename T>
```

Comment:
HOOO BOY, WE USIN TEMPLATES NOW, GOD HELP US 😈

Reply:
WE MAKING IT OUT OF C+ WITH THIS ONE! 🗣️ 🔈

Reply:
😨